w06-ruby-on-rails/reading-intro-to-class.md

```diff
@@ -0,0 +1,395 @@
+
+Notice that this `Car` class is spruced up with a new `@speed` instance variable and an `accelerate` instance method.
+
+```ruby
+class Car
+  attr_accessor :make, :color
+  @@count = 0
+
+  def initialize(color, make)
+    @speed = 0
+    @color = color
+    @make = make
+    @@count = @@count + 1
+  end
+
+  def self.count
+    @@count
+  end
+
+  def accelerate(change)
+		@speed += change
+	end
+end
+```
+
+
+### Subclass
+
+The syntax for inheritance uses `<` in the class definition.  Our pickup trucks will have another property that cars don't: the capacity of their truck beds (`@bed_capacity`). They'll also have a behavior that allows for riding in the back (`ride_in_back`).
+
+```ruby
+class Pickup < Car
+  attr_accessor :make, :color, :bed_capacity
+
+  def initialize(color, make, bed_capacity)
+    @speed = 0
+    @color = color
+    @make = make
+    @bed_capacity = bed_capacity
+    @@count = @@count + 1
+  end
+
+  def ride_in_back
+    puts "Bumpy but breezy!"
+  end
+end
+```
+
+Even though we didn't define the `accelerate` method again, a pickup truck will inherit the behavior from the `Car` class.
+
+```ruby
+truck_one = Pickup.new("red", "Ford", 100)
+truck_one.speed
+=> 0
+truck_one.accelerate(40)
+truck_one.speed
+=> 40
+```
+
+Inheritance doesn't go the other way, though -- new cars don't know how to use the `ride_in_back` behavior.
+
+```ruby
+focus = Car.new("green", "Ford")
+focus.ride_in_back
+```
+
+### Inheritance and Class Variables
+
+Class variables  in Ruby don't interact with inheritance in the way many people would expect.  All subclasses share the same class variable, so changing a class variable within a subclass changes the class variable for the base class and all other subclasses.  This can be good when, for instance, we want to update the total `Car` count whenever a new `Pickup` is created. However, `Pickup`'s `@@count` will always be equal to the total `Car` count.  
+
+This connection can cause lots of issues if it's not intended. For example, a `Vehicle` class might have an `@@num_wheels` variable that stores a "default" number of wheels for a vehicle, say 4.  If `Boat` is later created as a subclass of `Vehicle`, and `Boat`'s `@@num_wheels` is set to 0, then all vehicles will now have their number of wheels set to 0, even `Motorcylces`, `Cars`, `EighteenWheelers` and any other instances of `Vehicle` subclasses.
+
+A better pattern that fits this scenario uses "class instance variables". For the "class instance variable pattern", we create an instance variable for the shared data within the parent class itself (outside of a method definition). Then, we create a class getter method using within the parent class to access the class instance variable. Finally, we create a new version of the class instance variable for each subclass.  The data is no longer shared among multiple classes when we use this pattern. Let's see an example.
```

Comment:
class getter method using within the parent class   < -- awkward

Maybe preface this whole section with **tough**